### PR TITLE
[coreShadow] CBMC PROOF fix for Shadow_AssembleTopicString function

### PR DESCRIPTION
**Pull request by gshvang**
This PR contains CBMC PROOF fix for Shadow_AssembleTopicString function.

### CODE REVIEW

**Comment by gshvang** [comment]
Screen Shot 2021-05-17 at 11 53 39 AM

File: test/cbmc/proofs/Shadow_AssembleTopicString/Shadow_AssembleTopicString_harness.c [hunk 1]
```diff
@@ -49,7 +49,7 @@ void harness()
                                 thingNameLength,
                                 pShadowName,
                                 shadowNameLength,
-                                &( topicBuffer[ 0 ] ),
```
Comment (aggarw13) [inline test/cbmc/proofs/Shadow_AssembleTopicString/Shadow_AssembleTopicString_harness.c:52]: Can you provide the reason for the need of this this fix?

Comment (gshvang) [reply test/cbmc/proofs/Shadow_AssembleTopicString/Shadow_AssembleTopicString_harness.c:52]: If TopicBuffer is null then we cannot dereference the topicBuffer[ 0 ].

**Review by aggarw13 — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
 # Changelog for AWS IoT Device Shadow library
+## v1.1.1 (May 2021)
+
+### Other
+ - [#85](https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk/pull/85) Update CBMC proof for `Shadow_AssembleTopicString` API.
```
Comment (aggarw13) [inline CHANGELOG.md:5]: (Nit) It would be good to mention "Fix CBMC proof for..."

**Review by aggarw13 — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**